test/unit/transitions/multi_form_alerts.js

```diff
@@ -0,0 +1,384 @@
+const _ = require('underscore'),
+      config = require('../../../config'),
+      lineage = require('../../../lib/lineage'),
+      messages = require('../../../lib/messages'),
+      sinon = require('sinon').sandbox.create(),
+      transition = require('../../../transitions/multi_form_alerts'),
+      utils = require('../../../lib/utils'),
+      vm = require('vm');
+
+exports.tearDown = callback => {
+  sinon.restore();
+  callback();
+};
+
+exports.setUp = callback => {
+  sinon.stub(utils, 'getReportsWithinTimeWindow').returns(Promise.resolve(reports));
+  sinon.stub(lineage, 'fetchHydratedDocPromise').callsFake(id => {
+    if (id === doc._id) { return Promise.resolve(doc); }
+    if (id === reports[0]._id) { return Promise.resolve(hydratedReports[0]); }
+    if (id === reports[1]._id) { return Promise.resolve(hydratedReports[1]); }
+  });
```

Comment:
I don't like this pattern because it separates the mocking from the test, so when I come back and look at the test later I can never work out why it's working or not. The best way to do it is having something like the following in each test...

```js
sinon.stub(lineage, 'fetchHydratedDocPromise').withArgs(doc._id).returns(Promise.resolve(doc));
```

This makes it really clear how each unit is working.

Also by doing all the mocking in the `setUp` some tests may be passing incorrectly because you don't expect them to be calling this function, or calling it with these params, so you may miss an error.

And finally it means you can assign it to a variable and check things like the number of times it was called, which you can't do if you do it in the setUp and then throw it away.